components/script/dom/xmlhttprequest.rs

```diff
@@ -336,6 +339,18 @@ impl XMLHttpRequestMethods for XMLHttpRequest {
                         return Err(Error::InvalidAccess)
                     }
                 }
+
```

Comment:
We should be ascii-ifying the username by percent encoding first.

https://url.spec.whatwg.org/#set-the-username